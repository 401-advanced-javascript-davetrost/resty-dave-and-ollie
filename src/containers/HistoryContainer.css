.HistoryContainer {
  margin-top: 100px;

  h2 {
    font-family: Verdana;
    font-size: 16px;
  }
}