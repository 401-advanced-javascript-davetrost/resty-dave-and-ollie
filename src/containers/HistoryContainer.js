import React from 'react';
import PropTypes from 'prop-types';
import { HistoryItem } from '../components/HistoryItem';

export default function HistoryContainer({ items, handleHistorySelect }) {
  const historyItems = items.map(item => (
    <HistoryItem
      key={item._id}
      handleHistorySelect={handleHistorySelect}
      {...item} />
  ));

  return (
    <section>
      <h2>History</h2>
      <div>
        {historyItems}
      </div>
    </section>
  );
}

HistoryContainer.propTypes = {
  items: PropTypes.array.isRequired,
  // url: PropTypes.string.isRequired,
  // method: PropTypes.string.isRequired,
  // body: PropTypes.string.isRequired,
  handleHistorySelect: PropTypes.func.isRequired,
};
