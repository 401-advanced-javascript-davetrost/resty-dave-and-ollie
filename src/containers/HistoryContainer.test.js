import React from 'react';
import { shallow } from 'enzyme';
import HistoryContainer from './HistoryContainer';

describe('History Container component', () => {
  it('renders the container', () => {
    const sampleHistoryItem = {
      _id: 'xyzabc',
      url: 'random-api.com/api/characters',
      method: 'post',
      body: `{
        name: 'Garfield',
        color: 'Orange with black stripes'
      }`
    };
    const wrapper = shallow(<HistoryContainer 
      items={[sampleHistoryItem, sampleHistoryItem]}
      handleHistorySelect={() => true}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
