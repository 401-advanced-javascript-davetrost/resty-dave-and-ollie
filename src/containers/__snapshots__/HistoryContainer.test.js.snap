// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History Container component renders the container 1`] = `
<section
  className="HistoryContainer"
>
  <h2>
    History
  </h2>
  <div>
    <HistoryItem
      _id="xyzabc"
      body="{
        name: 'Garfield',
        color: 'Orange with black stripes'
      }"
      handleHistorySelect={[Function]}
      key="xyzabc"
      method="post"
      url="random-api.com/api/characters"
    />
    <HistoryItem
      _id="xyzabc"
      body="{
        name: 'Garfield',
        color: 'Orange with black stripes'
      }"
      handleHistorySelect={[Function]}
      key="xyzabc"
      method="post"
      url="random-api.com/api/characters"
    />
  </div>
</section>
`;
